packages/icon/package.py

```diff
@@ -1,294 +1,710 @@
-# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
-# Spack Project Developers. See the top-level COPYRIGHT file for details.
-#
-# SPDX-License-Identifier: (Apache-2.0 OR MIT)
-
 import os, subprocess
-from spack import *
-
-import llnl.util.tty as tty
+from collections import defaultdict
 
+from llnl.util import lang, filesystem
+from spack.util.environment import is_system_path, dump_environment
+from spack.util.executable import which_string
 
-class Icon(Package):
-    """The ICON modelling framework is a joint project between the
-    German Weather Service and the
-    Max Planck Institute for Meteorology for
-    developing a unified next-generation global numerical weather prediction and
-    climate modelling system. The ICON model has been introduced into DWD's
-    operational forecast system in January 2015."""
 
-    homepage = 'https://gitlab.dkrz.de/icon/icon'
-    url = 'https://gitlab.dkrz.de/icon/icon/-/archive/icon-2.6.2.2/icon-icon-2.6.2.2.tar.gz'
-    git = 'ssh://git@gitlab.dkrz.de/icon/icon.git'
+class Icon(AutotoolsPackage):
+    """Icosahedral Nonhydrostatic Weather and Climate Model."""
 
-    maintainers = ['jonasjucker', 'PanicSheep']
+    homepage = "https://code.mpimet.mpg.de/projects/iconpublic"
+    git = "git@github.com:C2SM/icon-exclaim.git"
```

Comment:
I think we should stick to what Sergey had: `git = 'ssh://git@gitlab.dkrz.de/icon/icon.git'`.
Adding an url would be nice for fetching versions: `url = 'https://gitlab.dkrz.de/icon/icon/-/archive/icon-2.6.5.1/icon-icon-2.6.5.1.tar.gz'`.

Reply:
Yes let's stick to sergeys version. Anyway we move to develop workflow using envs at least for Daint with this PR.